core/app/models/spree/product.rb

```diff
@@ -274,6 +274,10 @@ def find_variant_property_rule(option_value_ids)
       end
     end
 
+    def cache_key
+      super + "/prices/" + prices.currently_valid.pluck(:id).join("/")
```

Comment:
I thought about this but went for "the simplest thing that could possibly work". I do agree with digesting we'd be safe from key length issues. 

Something that's crossed my mind though: Should we maybe ask the currently used pricer make sure the correct cache key is used?